python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -369,6 +400,11 @@ def _resize_channel_if_needed(self, serialized_value: str):
         # include the size of the metadata, so we must account for the size of the
         # metadata explicitly.
         size = serialized_value.total_bytes + len(serialized_value.metadata)
+        if size > GRPC_LIMIT:
```

Comment:
Shouldn't we only throw this error if it's across nodes?

Reply:
Done.